File: leetcode/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:

        if len(t)< len(s):
            return False
        
        idx, jdx = 0,0

        while idx < len(s) and jdx< len(t):
            if s[idx] == t[jdx]:
                idx+=1
            jdx+=1
            
        
        if idx != len(s):
            return False
        else:
            return True




s,t = "bb", "b"

sol = Solution()
out = sol.isSubsequence(s,t)

print(out)

File: leetcode/arrays/392.go
package main

import "fmt"

func isSubsequence(s string, t string) bool {
	idx,jdx := 0, 0
	for idx<len(s) && jdx<len(t){
		if s[idx] == t[jdx]{
			idx++
			jdx++
		} else{
			jdx++
		}
	}

	if idx == len(s){
		return true
	}else{
		return false
	}

}

func main() {
	s, t := "", "ac"

	ans := isSubsequence(s, t)
	fmt.Println(ans)
}